### DISCUSSION TITLE
Windows Universal Target Not Working

### DISCUSSION THREAD

**Discussion by RuiCuco** [Kode/Kha#1458] [General]
I'm using VS Community 2022. The program stops with a breakpoint in a _debugbreak() call. The log says this before the breakpoint:

DXGI ERROR: IDXGIFactory::CreateSwapChain: No target window specified in DXGI_SWAP_CHAIN_DESC, and no window associated with owning factory. [ MISCELLANEOUS ERROR #6: ]

And then it continues:

A breakpoint instruction (__debugbreak() statement or a similar call) was executed in ek.exe.

Error: The specified module could not be found.

Am I doing something wrong? It works in HTML5 target.

**Comment by RobDangerous** [comment]
Haven't used the UWP target in forever, totally possible that it just broke. Please open an issue about it.

**Comment by RuiCuco** [comment]
The reason I'm interested is because I don't really know which target to use to publish in the Windows Store. Can you help?

**Comment by RobDangerous** [comment]
Nowadays that's actually done with the GDK (Microsoft themselves basically gave up on UWP for games). I have code for that but it's in the Xbox-target which is not public. Have to split it apart.

**Comment by RuiCuco** [comment]
I see. There are quite a few Windows targets in Kha. I tried the one that said simply 'Windows', but when running from VS Community there was an exception thrown (write access violation) in file constantbuffer.c.h here:

static void setMatrix4(uint8_t *constants, int offset, kinc_matrix4x4_t *value) {
	float *floats = (float *)(&constants[offset]);
	for (int y = 0; y < 4; ++y) {
		for (int x = 0; x < 4; ++x) {
			floats[x + y * 4] = kinc_matrix4x4_get(value, x, y); // ******    Here    ******
		}
	}
}

1 - Is this target functional, or did I do something wrong?

3 - Does this target allow publishing in Windows Store?

2 - Can you clean up the VS Code Kha extension, so that only relevant current targets remain?

**Comment by RobDangerous** [comment]
1. That target is functional. Can't tell what happened without seeing your source.
3. Maybe, not sure. They added the feature to publish regular Windows applications at some point but I don't know the current state of that.
2. I'm not aware of any irrelevant targets being in the extension.

**Comment by RuiCuco** [comment]
2. Ok, thanks.

3. Well, there is Android (Native) and Windows Universal, which doesn't appear to have much use. And what about WPF, what is that?

**Comment by RobDangerous** [comment]
3. Windows Universal is still useful for Xbox and WPF is a C#-build using Windows Presentation Foundation. But I did indeed forget to remove Android-Native - will be fixed, thanks.

**Comment by RuiCuco** [comment]
I see. Sadly, I couild not get khamake to build for WPF (this looks okay for the Windows Store, or am I wrong?). There are some errors:

Using Kha (b90792d0) from C:\Developer\Kha
Creating Kha project.
Exporting asset 1 of 3 (item1.png).
Exporting asset 2 of 3 (item2.png).
Exporting asset 3 of 3 (item3.png).
C:\Developer\Kha\Backends\WPF/kha/Image.hx:6: characters 8-34 : Type not found : kha.graphics4.hxsl.Globals
C:\Developer\Kha\Backends\WPF/system/windows/media/ImageSource.hx:5: characters 2-26 : Warning : (WExternWithExpr) Extern non-inline function may not have an expression
C:\Developer\Kha\Backends\WPF/system/windows/controls/MediaElement.hx:14: characters 2-32 : Warning : (WExternWithExpr) Extern non-inline function may not have an expression
C:\Developer\Kha\Backends\WPF/system/windows/controls/MediaElement.hx:16: characters 2-33 : Warning : (WExternWithExpr) Extern non-inline function may not have an expression
C:\Developer\Kha\Backends\WPF/system/windows/controls/MediaElement.hx:18: characters 2-34 : Warning : (WExternWithExpr) Extern non-inline function may not have an expression
C:\Developer\Kha\Backends\WPF/system/windows/controls/MediaElement.hx:20: characters 2-33 : Warning : (WExternWithExpr) Extern non-inline function may not have an expression
C:\Developer\Kha\Backends\WPF/kha/LoaderImpl.hx:52: characters 9-25 : kha.Kravur should be kha.Font
C:\Developer\Kha\Backends\WPF/kha/LoaderImpl.hx:52: characters 9-25 : ... For function argument ''
C:\Developer\Kha\Backends\WPF/kha/SystemImpl.hx:8: characters 8-15 : Type not found : kha.Key
C:\Developer\Kha\Sources/kha/System.hx:108: characters 28-36 : kha.Window -> Void should be () -> Void
C:\Developer\Kha\Sources/kha/System.hx:108: characters 28-36 : ... For function argument 'callback'

C:\Developer\Kha\Sources/kha/System.hx:197: characters 14-19 : Class<kha.SystemImpl> has no field login
C:\Developer\Kha\Sources/kha/System.hx:301: characters 3-38 : Void should be Bool
C:\Developer\Kha\Sources/kha/System.hx:366: characters 21-38 : Class<kha.SystemImpl> has no field automaticSafeZone
C:\Developer\Kha\Sources/kha/System.hx:370: characters 14-25 : Class<kha.SystemImpl> has no field setSafeZone
C:\Developer\Kha\Sources/kha/System.hx:374: characters 14-31 : Class<kha.SystemImpl> has no field unlockAchievement
C:\Developer\Kha\Backends\WPF/kha/wpf/Painter.hx:134: characters 3-16 : kha.Kravur should be kha.Font
C:\Developer\Kha\Backends\WPF/kha/wpf/Painter.hx:138: characters 3-42 : kha.Kravur should be kha.Font

Haxe compiler error.

**Comment by RobDangerous** [comment]
No, that's absolutely not suited for the Windows Store. The WPF target is very slow (and that can't be helped because of restrictions in WPF itself) and lacks many features - it's only useful for integrating some Kha-thing into a WPF application.
